### ISSUE TITLE
Cannot Activate Marketplace

### ISSUE THREAD

**Issue by tjapro** [opencart/opencart#9488]
**What version of OpenCart are you reporting this for?**
3.0.3.7

**Describe the bug**
When open the `route=marketplace/marketplace` give me this errors:
```
Notice: Trying to access array offset on value of type null in .../controller/marketplace/marketplace.php on line 167
Notice: Trying to access array offset on value of type null in .../controller/marketplace/marketplace.php on line 205
Notice: Trying to access array offset on value of type null in .../controller/marketplace/marketplace.php on line 222
```
And when introduce the API user and secret, it gives me ok, but nothing appears.

**To Reproduce**
1. Go to admin and `route=marketplace/marketplace`
2. See error

**Expected behavior**
No errors or notices and a displaying marketplace.

**Screenshots / Screen recordings**
**Server / Test environment (please complete the following information):**
- VPS
- CentOS 7
- PHP 8.0.3
- Nginx 1.16.1
- Chrome Version 89.0.4389.90

**Additional context**
I opened a similar issue - now closed - (https://github.com/opencart/opencart/issues/9161), but the enviroment isn't the same.

**Comment by prabhatkumaroc082** [comment]
Hello @tjapro 
Kindly try to apply this change https://github.com/opencart/opencart/commit/b24c1145faf31e5fa88626ce3baea2c3d276954c

**Comment by tjapro** [comment]
Sorry, but there are several strings `marketplace/api` inside curl OPENCART_SERVER and inside $json ifs ...
I made a fast search for that at https://github.com/opencart/opencart/commit/b24c1145faf31e5fa88626ce3baea2c3d276954c?branch=b24c1145faf31e5fa88626ce3baea2c3d276954c&diff=split#diff-ab66257ea1b8384ae888f8496bbedd1be3704f14355395d18ae9c5a78d1d558c

Is to change all of them?

**Comment by prabhatkumaroc082** [comment]
> I made a fast search for that at [b24c114?branch=b24c1145faf31e5fa88626ce3baea2c3d276954c&diff=split#diff-ab66257ea1b8384ae888f8496bbedd1be3704f14355395d18ae9c5a78d1d558c](https://github.com/opencart/opencart/commit/b24c1145faf31e5fa88626ce3baea2c3d276954c?branch=b24c1145faf31e5fa88626ce3baea2c3d276954c&diff=split#diff-ab66257ea1b8384ae888f8496bbedd1be3704f14355395d18ae9c5a78d1d558c)

Yes, kindly apply this change and let me know if you face any other issues.

**Comment by tjapro** [comment]
I made that changes as asked.
The errors are the following:
```
Notice: Trying to access array offset on value of type null in /var/www/.../admin/controller/marketplace/marketplace.php on line 167
Notice: Trying to access array offset on value of type null in /var/www/.../admin/controller/marketplace/marketplace.php on line 205
Notice: Trying to access array offset on value of type null in /var/www/.../admin/controller/marketplace/marketplace.php on line 222
```

**Comment by prabhatkumaroc082** [comment]
@tjapro 
I have validated the result so you must not face notices now, kindly use this file https://github.com/opencart/opencart/blob/3.0.x.x_Maintenance/upload/admin/controller/marketplace/marketplace.php code for the latest 3.0.3.6 or 3.0.3.7 stores

If you are not getting results then make sure your store version is 3.0.3.7 because v3.0.3.8 is not resulting.

**Comment by tjapro** [comment]
I'm using 3.0.3.7, the latest official OC release.

Congratulations!!! It accepts my marketplace credentials without problems or notices!

But just two questions:
- Why I search throught marketplace and it doesn't give me any results?
![without results](https://user-images.githubusercontent.com/7900361/113361736-8302f380-9344-11eb-91e7-80890bc490e1.png)
- Will be this changes be part of the 3.0.3.8 release?

**Comment by prabhatkumaroc082** [comment]
> I'm using 3.0.3.7, the latest official OC release.
> 
> Congratulations!!! It accepts (I guess) my marketplace credentials without problems or notices!
> 
> But just two questions:
> 
> * Why I search throught marketplace and it doesn't give me any results?
>   ![without results](https://user-images.githubusercontent.com/7900361/113361736-8302f380-9344-11eb-91e7-80890bc490e1.png)
> * Will be this changes be part of the 3.0.3.8 release?

After applying file https://github.com/opencart/opencart/blob/3.0.x.x_Maintenance/upload/admin/controller/marketplace/marketplace.php changes, any issue is not faced.


> If you are not getting results then make sure your store version is 3.0.3.7 because v3.0.3.8 is not resulting.
Please verify your version https://prnt.sc/114lf8f as well.

**Comment by tjapro** [comment]
I changed the all file.
Strange: I installed the version 3.0.3.7 from composer, but OC admin give me 3.0.3.6 ....
I'll see with more detail.

**Comment by prabhatkumaroc082** [comment]
> Strange: I installed the version 3.0.3.7 from composer, but OC admin give me 3.0.3.6 ....

I will test it and will fix it for v3.0.3.8 if any issue.

**Comment by Webkul-Opencart** [comment]
@tjapro I have not found any issue for the version 3.0.3.7.
